WordPress/gutenberg
# Initialize new post with auto-draft

This pull request seeks to initialize a new post draft with an auto-draft, matching the behavior of the current post editor, for a few reasons:

- Simplifies initialization steps by being able to assume a post object always exists
- Enables "Uploaded to" media library behavior
- Will allow access to correct default post values (comment status, ping status, category, format, etc)
   - I'm hoping to use this for more accurate dirty detection between edited and saved post values

__Testing instructions:__

Verify that for all intents and purposes, a new post looks and acts as though it has not yet been saved. This includes: No save button until edits made, no post permalink when focusing title, no Trash button, no revisions (nor revisions network requests), no warnings when navigating away without edits, updated URL after first save.
